### ISSUE TITLE
Notebook cell code not visible

### ISSUE THREAD

**Issue by ieglonew01f** [jupyterlab/jupyterlab#7072]
## Description

When a user opens an existing notebook which has some code inside it and while this notebook is loading if the user switches to a different notebook which was already present in the main area, upon switching back to the recently opened notebook, it doesn't show the code of that notebook.

## GIF
![jpy3](https://user-images.githubusercontent.com/9274615/63514171-9f527380-c505-11e9-96e9-9ef2fc2e35d5.gif)

## Steps to reproduce

1. Ensure that you have a notebook open already in the main area.
2. Ensure that you have another notebook which has some code in it.
3. Open up the chrome debugger and set network throttling to 3G or 2G.
4. Now open the other notebook.
5. Before the notebook finish loading its content switch back to the previous notebook which was open
6. Now switch back to the recently loaded notebook.
7. Notice that the cell editor code is not visible or missing or has not loaded.

## Expected behavior

It should load the cell editor code even when the notebook which is loading is not currently in the viewport.

## Context

- Operating System and version: OSX Mojave
- Browser and version: Chrome 76.0.3809.100 (Official Build) (64-bit)
- JupyterLab version: 1.0.1
- Jupyter version: 4.4.0

**Comment by ieglonew01f** [comment]
One more thing I've observed is that the content of the code editor is present in the DOM, but it's hidden.
<img width="1029" alt="Screenshot 2019-08-23 at 11 41 40 AM" src="https://user-images.githubusercontent.com/9274615/63570486-0bcb8200-c59b-11e9-9dd7-58e5b8146c97.png">

**Comment by ieglonew01f** [comment]
Found the issue, looks like its an issue with codemirror itself
https://stackoverflow.com/questions/8349571/codemirror-editor-is-not-loading-content-until-clicked

**Comment by ieglonew01f** [comment]
Also there is a libary by codemirror just for this problem.
https://codemirror.net/doc/manual.html#addon_autorefresh

> This addon can be useful when initializing an editor in a hidden DOM node, in cases where it is difficult to call refresh when the editor becomes visible. It defines an option autoRefresh which you can set to true to ensure that, if the editor wasn't visible on initialization, it will be refreshed the first time it becomes visible. This is done by polling every 250 milliseconds (you can pass a value like {delay: 500} as the option value to configure this). Note that this addon will only refresh the editor once when it first becomes visible, and won't take care of further restyling and resizing.

**Comment by vidartf** [comment]
I would have a look in the cell widget's `onShow` handler, and ensure that it refreshes on the first show after its creation.

**Comment by Prakhar834** [comment]
> I would have a look in the cell widget's `onShow` handler, and ensure that it refreshes on the first show after its creation.

There is an option for updateEditorOnShow in cell widget. By default it set as false, issue can be resolved by setting it True. It works by calling refresh method for cells when codeEditor is visible. Not sure of what cons it may have apart from an overhead of refresh.

**Comment by vidartf** [comment]
> Not sure of what cons it may have apart from an overhead of refresh. 

Yes, but that overhead is quite significant! There is a huge issue on notebook performance already in e.g. #4292 .

**Comment by bharat-ics** [comment]
Hi if the issue has been resolved, would it be possible to detail the steps necessary to duplicate the solution. I have been facing this issue as well. Thanks

**Comment by veve90** [comment]
I have the same issue, are there any steps for solving this issue ?
